### ISSUE TITLE
feat: arguments should be supported in both str or native format

### ISSUE THREAD

**Issue by yxliang01** [crytic/crytic-compile#108]
Many arguments are expected to be serialized `str` e.g. `solc_args`. However, crytic-compile will often need to deserialize them and also serialize again.

While it works perfect for user inputs (e.g. arguments taken from CLI), it would be a unnecessary overhead when CryticCompile is used as a library that developers need to serialize for no use (because soon will be deserized in CryticCompile).

**Comment by montyly** [comment]
Hi @yxliang01 ,

Do you have an example that you could share? 

If I understand correctly, you want to give to `solc_args` either a string, or a list of strings, is that correct?

**Comment by yxliang01** [comment]
@montyly Yes, your understanding is correct. For my projects, the arguments are constructed naturally in a form of list of strings. But, to let crytic-compile use, I have to join them. This effort looks useless because crytic-compile will split it again.
